node-uniffi/src/types.rs

```diff
@@ -0,0 +1,516 @@
+    str::FromStr,
+    time::{Duration, SystemTime},
+};
+use uniffi::Record;
+
+use lumina_node::store::RedbStore;
+
+use crate::{error::Result, LuminaError};
+
+#[cfg(target_os = "ios")]
+use directories::ProjectDirs;
+
+#[cfg(target_os = "ios")]
+/// Returns the platform-specific base path for storing on iOS.
+fn get_base_path_impl() -> Result<PathBuf> {
+    if let Some(proj_dirs) = ProjectDirs::from("com", "example", "Lumina") {
+        Ok(proj_dirs.data_dir().to_path_buf())
+    } else {
+        Err(LuminaError::StorageError {
+            msg: "Could not determine a platform-specific data directory".to_string(),
+        })
+    }
+}
+
+#[cfg(target_os = "android")]
+/// Returns the platform-specific base path for storing on Android.
+///
+/// On Android, this function attempts to read the `LUMINA_DATA_DIR` environment variable.
+/// If `LUMINA_DATA_DIR` is not set, it falls back to `/data/data/com.example.lumina/files`.
+fn get_base_path_impl() -> Result<PathBuf> {
+    match std::env::var("LUMINA_DATA_DIR") {
+        Ok(dir) => Ok(PathBuf::from(dir)),
+        Err(_) => {
+            let fallback = "/data/data/com.example.lumina/files";
+            Ok(PathBuf::from(fallback))
+        }
+    }
+}
+
+#[cfg(not(any(target_os = "ios", target_os = "android")))]
+/// Returns an error for unsupported platforms.
+fn get_base_path_impl() -> Result<PathBuf> {
+    Err(LuminaError::StorageError {
+        msg: "Unsupported platform".to_string(),
+    })
+}
+
+/// Returns the platform-specific base path for storing Lumina data.
+///
+/// The function determines the base path based on the target operating system:
+/// - **iOS**: `~/Library/Application Support/lumina`
+/// - **Android**: Value of the `LUMINA_DATA_DIR` environment variable
+/// - **Other platforms**: Returns an error indicating unsupported platform.
+fn get_base_path() -> Result<PathBuf> {
+    get_base_path_impl()
+}
+
+/// Configuration options for the Lumina node
+#[derive(Debug, Clone, Record)]
+pub struct NodeStartConfig {
+    /// Network to connect to
+    pub network: network::Network,
+    /// Custom list of bootstrap peers to connect to.
+    /// If None, uses the canonical bootnodes for the network.
+    pub bootnodes: Option<Vec<String>>,
+    /// Custom syncing window in seconds. Default is 30 days.
+    pub syncing_window_secs: Option<u32>,
+    /// Custom pruning delay after syncing window in seconds. Default is 1 hour.
+    pub pruning_delay_secs: Option<u32>,
+    /// Maximum number of headers in batch while syncing. Default is 128.
+    pub batch_size: Option<u64>,
+    /// Optional Set the keypair to be used as Node's identity. If None, generates a new Ed25519 keypair.
+    pub ed25519_secret_key_bytes: Option<Vec<u8>>,
+}
+
+impl NodeStartConfig {
+    /// Convert into NodeBuilder for the implementation
+    pub(crate) async fn into_node_builder(self) -> Result<NodeBuilder<RedbBlockstore, RedbStore>> {
+        let base_path = get_base_path()?;
```

Comment:
IMO, this is too high-level api for a library. I think it would be better for the user to select the base path thought `NodeStartConfig`.

This removes the responsibility from us to figure out the project dir or make use of environment variables within the library.